### DISCUSSION TITLE
Results of a model for my native language

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#235] [Archived]
<USER>>>> Immabot<br/> <DATE> [July 15, 2020, 8:34am]<br/> <br/> and community,<br/> <br/> I am currently training a model for my native language. My dataset<br/> includes around 30 hours of audio.<br/> <br/> Here's the config file -<br/> <br/> 'model': 'Tacotron2', *slash* <br/> 'run_name': 'ljspeech', *slash* <br/> 'run_description': 'tacotron2',<br/> <br/> // AUDIO PARAMETERS<br/> 'audio':{<br/> // stft parameters<br/> 'num_freq': 1025, // number of stft frequency levels. Size of the linear spectogram frame.<br/> 'win_length': 1024, // stft window length in ms.<br/> 'hop_length': 256, // stft window hop-lengh in ms.<br/> 'frame_length_ms': null, // stft window length in ms.If null, 'win_length' is used.<br/> 'frame_shift_ms': null, // stft window hop-lengh in ms. If null, 'hop_length' is used.<br/> <br/> // Audio processing parameters<br/> 'sample_rate': 22050, // DATASET-RELATED: wav sample-rate. If different than the original data, it is resampled.<br/> 'preemphasis': 0.0, // pre-emphasis to reduce spec noise and make it more structured. If 0.0, no -pre-emphasis.<br/> 'ref_level_db': 20, // reference level db, theoretically 20db is the sound of air.<br/> <br/> // Silence trimming<br/> 'do_trim_silence': true,// enable trimming of slience of audio as you load it. LJspeech (false), TWEB (false), Nancy (true)<br/> 'trim_db': 60, // threshold for timming silence. Set this according to your dataset.<br/> <br/> // Griffin-Lim<br/> 'power': 1.5, // value to sharpen wav signals after GL algorithm.<br/> 'griffin_lim_iters': 60,// #griffin-lim iterations. 30-60 is a good range. Larger the value, slower the generation.<br/> <br/> // MelSpectrogram parameters<br/> 'num_mels': 80, // size of the mel spec frame.<br/> 'mel_fmin': 0.0, // minimum freq level for mel-spec. ~50 for male and ~95 for female voices. Tune for dataset!!<br/> 'mel_fmax': 8000.0, // maximum freq level for mel-spec. Tune for dataset!!<br/> <br/> // Normalization parameters<br/> 'signal_norm': true, // normalize spec values. Mean-Var normalization if 'stats_path' is defined otherwise range normalization defined by the other params.<br/> 'min_level_db': -100, // lower bound for normalization<br/> 'symmetric_norm': true, // move normalization to range [-1, 1]<br/> 'max_norm': 4.0, // scale normalization to range [-max_norm, max_norm] or [0, max_norm]<br/> 'clip_norm': true, // clip normalized values into the range.<br/> 'stats_path': null // DO NOT USE WITH MULTI_SPEAKER MODEL. scaler stats file computed by 'compute_statistics.py'. If it is defined, mean-std based notmalization is used and other normalization params are ignored<br/> },<br/> <br/> // DISTRIBUTED TRAINING<br/> 'distributed':{<br/> 'backend': 'nccl',<br/> 'url': 'tcp: *slash* / *slash* /localhost:54321'<br/> },<br/> <br/> 'reinit_layers': [], // give a list of layer names to restore from the given checkpoint. If not defined, it reloads all heuristically matching layers.<br/> <br/> // TRAINING<br/> 'batch_size': 32, // Batch size for training. Lower values than 32 might cause hard to learn attention. It is overwritten by 'gradual_training'.<br/> 'eval_batch_size':16,<br/> 'r': 7, // Number of decoder frames to predict per iteration. Set the initial values if gradual training is enabled.<br/> 'gradual_training': [[0, 7, 64], [1, 5, 32], [50000, 5, 32], [130000, 5, 32], [290000, 5, 32]], //set gradual training steps [first_step, r, batch_size]. If it is null, gradual training is disabled. For Tacotron, you might need to reduce the 'batch_size' as you proceeed.<br/> 'loss_masking': true, // enable / disable loss masking against the sequence padding.<br/> 'ga_alpha': 10.0, // weight for guided attention loss. If > 0, guided attention is enabled.<br/> <br/> // VALIDATION<br/> 'run_eval': true,<br/> 'test_delay_epochs': 0, //Until attention is aligned, testing only wastes computation time.<br/> 'test_sentences_file': null, // set a file to load sentences to be used for testing. If it is null then we use default english sentences.<br/> <br/> // OPTIMIZER<br/> 'noam_schedule': false, // use noam warmup and lr schedule.<br/> 'grad_clip': 1.0, // upper limit for gradients for clipping.<br/> 'epochs': 1000, // total number of epochs to train.<br/> 'lr': 0.0001, // Initial learning rate. If Noam decay is active, maximum learning rate.<br/> 'wd': 0.000001, // Weight decay weight.<br/> 'warmup_steps': 4000, // Noam decay steps to increase the learning rate from 0 to 'lr'<br/> 'seq_len_norm': false, // Normalize eash sample loss with its length to alleviate imbalanced datasets. Use it if your dataset is small or has skewed distribution of sequence lengths.<br/> <br/> // TACOTRON PRENET<br/> 'memory_size': -1, // ONLY TACOTRON - size of the memory queue used fro storing last decoder predictions for auto-regression. If < 0, memory queue is disabled and decoder only uses the last prediction frame.<br/> 'prenet_type': 'original', // 'original' or 'bn'.<br/> 'prenet_dropout': true, // enable/disable dropout at prenet.<br/> <br/> // ATTENTION<br/> 'attention_type': 'original', // 'original' or 'graves'<br/> 'attention_heads': 4, // number of attention heads (only for 'graves')<br/> 'attention_norm': 'sigmoid', // softmax or sigmoid. Suggested to use softmax for Tacotron2 and sigmoid for Tacotron.<br/> 'windowing': false, // Enables attention windowing. Used only in eval mode.<br/> 'use_forward_attn': false, // if it uses forward attention. In general, it aligns faster.<br/> 'forward_attn_mask': false, // Additional masking forcing monotonicity only in eval mode.<br/> 'transition_agent': false, // enable/disable transition agent of forward attention.<br/> 'location_attn': true, // enable_disable location sensitive attention. It is enabled for TACOTRON by default.<br/> 'bidirectional_decoder': false, // use https://arxiv.org/abs/1907.09006. Use it, if attention does not work well with your dataset.<br/> <br/> // STOPNET<br/> 'stopnet': true, // Train stopnet predicting the end of synthesis.<br/> 'separate_stopnet': true, // Train stopnet seperately if 'stopnet==true'. It prevents stopnet loss to influence the rest of the model. It causes a better model, but it trains SLOWER.<br/> <br/> // TENSORBOARD and LOGGING<br/> 'print_step': 25, // Number of steps to log traning on console.<br/> 'print_eval': false, // If True, it prints intermediate loss values in evalulation.<br/> 'save_step': 10000, // Number of training steps expected to save traninpg stats and checkpoints.<br/> 'checkpoint': true, // If true, it saves checkpoints per 'save_step'<br/> 'tb_model_param_stats': false, // true, plots param stats per layer on tensorboard. Might be memory consuming, but good for debugging.<br/> <br/> // DATA LOADING<br/> 'text_cleaner': 'basic_cleaners',<br/> 'enable_eos_bos_chars': false, // enable/disable beginning of sentence and end of sentence chars.<br/> 'num_loader_workers': 4, // number of training data loader processes. Don't set it too big. 4-8 are good values.<br/> 'num_val_loader_workers': 4, // number of evaluation data loader processes.<br/> 'batch_group_size': 0, //Number of batches to shuffle after bucketing.<br/> 'min_seq_len': 6, // DATASET-RELATED: minimum text length to use in training<br/> 'max_seq_len': 153, // DATASET-RELATED: maximum text length<br/> <br/> // PATHS<br/> 'output_path': '/content/drive/My Drive/output/',<br/> <br/> // PHONEMES<br/> 'phoneme_cache_path': '/media/erogol/data_ssd2/mozilla_us_phonemes_3', // phoneme computation is slow, therefore, it caches results in the given folder.<br/> 'use_phonemes': false, // use phonemes instead of raw characters. It is suggested for better pronounciation.<br/> 'phoneme_language': 'en-us', // depending on your target language, pick one from https://github.com/bootphon/phonemizer#languages<br/> <br/> // MULTI-SPEAKER and GST<br/> 'use_speaker_embedding': false, // use speaker embedding to enable multi-speaker learning.<br/> 'style_wav_for_test': null, // path to style wav file to be used in TacotronGST inference.<br/> 'use_gst': false, // TACOTRON ONLY: use global style tokens<br/> <br/> // DATASETS<br/> 'datasets': // List of datasets. They all merged and they get different speaker_ids.<br/> [<br/> {<br/> 'name': 'ljspeech',<br/> 'path': '/content/',<br/> 'meta_file_train': 'metadata.csv',<br/> 'meta_file_val': null<br/> }<br/> ]<br/> <br/> Tensorboard logs -<br/> <br/> 13-38-57.filename}[1920×1080 172<br/> <br/> *slash* <br/> <br/> 13-39-04.filename}[1920×1080 181<br/> <br/> *slash* <br/> <br/> 13-39-12.filename}[1920×1080 172<br/> <br/> *slash* <br/> <br/> 13-39-38.filename}[1920×1080 289<br/> <br/> *slash* <br/> <br/> 13-39-27.filename}[1920×1080 300<br/> <br/> I previously trained a model with gradual training for around 300k<br/> steps. But the test sentences had a trailing silence in the end(decoder<br/> stopped with *slash* 'max_decoder_steps. Although I'm not sure if these two are<br/> related). Should i continue with r = 5 and if yes for how long? The<br/> current audio quality of the synthesized test sentences is kinda okay<br/> but I'm not sure if it's improving with further training. Also am I<br/> missing something which is causing the trailing silence at the end? And<br/> can you help me in interpreting the tensorboard logs?<br/> <br/> Any help is appreciated.<br/> <br/> Thanks in advance!<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/results-of-a-model-for-my-native-language]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> georroussos<br/> <DATE> [July 15, 2020, 11:08am]<br/> <br/> Hi, what is the language? I would try a run with phonemes and forward<br/> attention. Is your dataset clean of trailing silences?<br/> <br/> [Archived Post]<br/>
